S3: panic: runtime error: slice bounds out of range

Complete backup n00b here. Previously just used rsync to a local disk.

Installation went very smoothly (Debian linux):

restic 0.1.0 (v0.1.0-348-geccbcb7)
compiled at 2016-02-10 00:48:37 with go1.5.3

As a test I used a 700MB directory, mainly small text files with a lot of duplication and a few big binaries up to 150MB. Backing up to the local disk worked OK, so I tried S3, which was my reason for looking at this.

It died at about 66%. Restarted several times, each time initially showing about 25%. Sorry I didn't capture all the output. But now it fails as soon as I enter the password:

scanned 229 directories, 8768 files in 0:00
panic: runtime error: slice bounds out of range

My connection is around 1Mbit/s up and 30Mbit/s down. Mostly the upload was going about twice this rate, I assume because of many near-identical files.

cheers
Nigel

--- a/cmd/restic/cmd_backup.go
+++ b/cmd/restic/cmd_backup.go
@@ -153,9 +153,13 @@ func (cmd CmdBackup) newArchiveProgress(todo restic.Stat) *restic.Progress {
 
 		w, _, err := terminal.GetSize(int(os.Stdout.Fd()))
 		if err == nil {
-			if len(status1)+len(status2) > w {
-				max := w - len(status2) - 4
-				status1 = status1[:max] + "... "
+			maxlen := w - len(status2)
+
+			if maxlen < 4 {
+				status1 = ""
+			} else if len(status1) > maxlen {
+				status1 = status1[:maxlen-4]
+				status1 += "... "
 			}
 		}
 

